src/libos/src/events/notifier.rs

```diff
@@ -0,0 +1,62 @@
+use std::any::Any;
+use std::sync::Weak;
+
+use super::{Event, EventFilter, Observer};
+use crate::prelude::*;
+
+// TODO: how to prevent deadlock or infinite loop during event propagation?
```

Comment:
Can you future explain all the TODOs in this commit? What is the influence and how should it be done in the future? This commit acts as an enhancement. It should not introduce new problems.

Reply:
This is a good question. I will explain a bit more in a proper place.

Reply:
Comments are added to `Observer::on_event` to explain the avoidance of deadlock.